pallets/asset/src/lib.rs

```diff
@@ -1929,6 +1944,12 @@ impl<T: Config> Module<T> {
 
         Self::ensure_granular(&ticker, value)?;
 
+        // Ensures the token is fungible
+        ensure!(
+            Tokens::get(&ticker).asset_type.is_fungible(),
+            Error::<T>::UnexpectedNonFungibleToken
+        );
```

Comment:
The `ensure_granular` check also read the `Tokens` storage for the `divisible` flag.  These two checks could be combined to avoid the extra read.